packages/react-reconciler/src/SchedulingProfiling.js

```diff
@@ -0,0 +1,189 @@
+      const id = getWakeableID(wakeable);
+      const componentStack = getStackByFiberInDevAndProd(fiber) || '';
+      // TODO (brian) Generate and store temporary ID so DevTools can match up a component stack later.
+      performance.mark(
+        `--suspense-suspend-${componentName}-${id}-${componentStack}`,
+      );
+      wakeable.then(
+        () =>
+          performance.mark(
+            `--suspense-resolved-${componentName}-${id}-${componentStack}`,
+          ),
+        () =>
+          performance.mark(
+            `--suspense-rejected-${componentName}-${id}-${componentStack}`,
+          ),
+      );
+    }
+  }
+}
+
+export function markLayoutEffectsStarted(lanes: Lanes): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark(`--layout-effects-start-${formatLanes(lanes)}`);
+    }
+  }
+}
+
+export function markLayoutEffectsStopped(): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark('--layout-effects-stop');
+    }
+  }
+}
+
+export function markPassiveEffectsStarted(lanes: Lanes): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark(`--passive-effects-start-${formatLanes(lanes)}`);
+    }
+  }
+}
+
+export function markPassiveEffectsStopped(): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark('--passive-effects-stop');
+    }
+  }
+}
+
+export function markRenderStarted(lanes: Lanes): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark(`--render-start-${formatLanes(lanes)}`);
+    }
+  }
+}
+
+export function markRenderYielded(): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark('--render-yield');
+    }
+  }
+}
+
+export function markRenderStopped(): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark('--render-stop');
+    }
+  }
+}
+
+export function markRenderScheduled(
+  componentName: string,
+  fiber: Fiber,
+  lane: Lane,
+): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      const componentStack = getStackByFiberInDevAndProd(fiber) || '';
+      // TODO (brian) Generate and store temporary ID so DevTools can match up a component stack later.
+      performance.mark(
+        `--schedule-render-${componentName}-${formatLanes(
+          lane,
+        )}-${componentStack}`,
+      );
+    }
+  }
+}
+
+export function markForceUpdateScheduled(
+  componentName: string,
+  fiber: Fiber,
+  lane: Lane,
+): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      const componentStack = getStackByFiberInDevAndProd(fiber) || '';
+      // TODO (brian) Generate and store temporary ID so DevTools can match up a component stack later.
+      performance.mark(
+        `--schedule-forced-update-${componentName}-${formatLanes(
```

Comment:
nit: Maybe `markForceUpdateScheduled` and `markStateUpdateScheduled` should align with the other mark methods and log the lane info first (followed by component name and stack)? Seems like that consistency would be nice.